community-connectors/google-fit/Connector.js

```diff
@@ -0,0 +1,572 @@
+  }
+
+}
+
+/**
+ * Implements the callback for the OAuth authorization flow.
+ * 
+ * @param {Object} request the oauth callback request
+ */
+connector.authCallback = function(request) {
+  var fit = new GoogleFit(OAUTH_CLIENT_ID, OAUTH_CLIENT_SECRET);
+  var service = fit.getService();
+
+  var authorized = service.handleCallback(request);
+  if (authorized) {
+    return HtmlService.createHtmlOutput('Success! You can close this tab.');
+  } else {
+    return HtmlService.createHtmlOutput('Denied. You can close this tab');
+  };
+};
+
+/**
+ * An authentication type object.
+ * @typedef {Object} AuthType
+ * @property {string} type - One of "NONE" or "OAUTH2"
+ * @see {@link https://developers.google.com/datastudio/connector/reference#authtype|AuthType reference}
+ */
+
+/**
+ * Used by DataStudio to get the authorization type used by this connector.
+ * 
+ * @return {AuthType} an object representing the auth type.
+ */
+connector.getAuthType = function() {
+  var response = {
+    "type": "OAUTH2"
+  };
+  return response;
+};
+
+/**
+ * Used by DataStudio to get whether the connector is currently authorized.
+ *
+ * @return {boolean} whether the connector is currently authorized.
+ * @see {@link https://developers.google.com/datastudio/connector/reference#isauthvalid|isAuthValid reference}
+ */
+connector.isAuthValid = function() {
+  var fit = new GoogleFit(OAUTH_CLIENT_ID, OAUTH_CLIENT_SECRET);
+  var service = fit.getService();
+  if (service == null) {
+    return false;
+  }
+  return service.hasAccess();
+};
+
+/**
+ * Used by DataStudio to initiate OAuth 2.0 flow for a 3rd-party service.
+ *
+ * @return {string} Returns the authorization URL for the 3rd-party service. The authorization URL will be presented to the user to initiate the OAuth 2.0 flow to grant access to the 3rd-party service.
+ * @see {@link https://developers.google.com/datastudio/connector/reference#get3pauthorizationurls|get3PAuthorizationUrls reference}
+ */
+connector.get3PAuthorizationUrls = function() {
+  var fit = new GoogleFit(OAUTH_CLIENT_ID, OAUTH_CLIENT_SECRET);
+  var service = fit.getService();
+  if (service == null) {
+    return '';
+  }
+  return service.getAuthorizationUrl();
+};
+
+/**
+ * Used by DataStudio to reset authorization for the connector.
+ *
+ * @see {@link https://developers.google.com/datastudio/connector/reference#resetauth|resetAuth reference}
+ */
+connector.resetAuth = function() {
+  var fit = new GoogleFit(OAUTH_CLIENT_ID, OAUTH_CLIENT_SECRET);
+  var service = fit.getService();
+  service.reset();
+}
+
+/**
+ * This checks whether the current user is an admin user of the connector.
+ *
+ * @returns {boolean} Returns true if the current authenticated user at the time
+ * of function execution is an admin user of the connector. If the function is
+ * omitted or if it returns false, then the current user will not be considered
+ * an admin user of the connector.
+ */
+connector.isAdminUser = function () {
+  return true;
```

Comment:
Ideally, this would be false when published. We don't need logging to occur for non-developer users.

Reply:
I just removed this method since it's not really serving any purpose.